src/main/java/net/masterthought/cucumber/json/deserializers/StepDeserializer.java

```diff
@@ -0,0 +1,65 @@
+package net.masterthought.cucumber.json.deserializers;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import net.masterthought.cucumber.json.DocString;
+import net.masterthought.cucumber.json.Embedding;
+import net.masterthought.cucumber.json.Match;
+import net.masterthought.cucumber.json.Output;
+import net.masterthought.cucumber.json.Result;
+import net.masterthought.cucumber.json.Row;
+import net.masterthought.cucumber.json.Step;
+
+public class StepDeserializer extends JsonDeserializer<Step> {
+
+	@Override
+	public Step deserialize(JsonParser parser, DeserializationContext context) throws IOException {
```

Comment:
instead of providing deserializer that makes code ugly, maybe try to provide additional method such as getOutputs() that wraps it? Then the model stays the  same as iti s in json file and onlyl view is using method for displaying content in different way